src/schemas/activity.ts

```diff
@@ -0,0 +1,106 @@
+import { z } from "zod";
+
+export const activitySchema = z
+  .object({
+    Type: z.string().optional(),
+    DueDate: z.string(),
+    Subject: z.string(),
+    Status: z.string(),
+    AssignedTo: z.string(),
+    ActivityDateTime: z.string(),
+    EndDateTime: z.string(),
+    EndDate: z.string(),
+    Location: z.string(),
+    To: z.string(),
+    Bcc: z.string(),
+    Body: z.string(),
+    Comments: z.string(),
+    Description: z.string(),
+  })
+  .transform((obj) => {
+    const newObj = {
+      ...obj,
+      [obj.Type === "Event" ? "EventSubtype" : "TaskSubtype"]: obj.Type,
+    };
+
+    delete newObj.Type;
+    return newObj;
+  });
+
+export const TaskSchema = z
+  .object({
+    Type: z.string().optional(),
+    ActivityDate: z.string(),
+    Subject: z.string().min(1),
+    Status: z.string().min(1),
+    OwnerId: z.string(),
+    WhoId: z.string().optional(),
+    AccountId: z.string().optional(),
+  })
+  .transform((obj) => {
+    const newObj = {
+      ...obj,
+      TaskSubtype: "Task",
+    };
+
+    return newObj;
+  });
+
+export const EventSchema = z
```

Comment:
This validation schema should come from the metadata of the sObject - i.e. field names, which fields are required, etc. (as above)